sdk/typescript/test/e2e/data/coin_metadata/sources/coin_metadata.move

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+module coin_metadata::test {
+    use sui::coin::{Self, SuiCoinRegistry};
+    use sui::transfer;
+    use sui::tx_context::TxContext;
+
+    /// Name of the coin. By convention, this type has the same name as its parent module
+    /// and has no fields. The full type of the coin defined by this module will be `COIN<TEST>`.
+    struct TEST has drop {}
+
+    // !!! Chris: cannot pass in SuiCoinRegistry since module
+    // initializer can not have extra parameters
```

Comment:
Unfortunately we can't pass in `SuiCoinRegistry` here for `coin::create_currency` due to module initializer constraints